File: web/src/test/java/XmlRenderingTest.java
import be.geoffrey.xmlweave.core.usecase.ElementRepresentation;
import be.geoffrey.xmlweave.service.XmlRenderer;
import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class XmlRenderingTest {

    @Test
    public void simpleRendering() {

        XmlRenderer renderer = new XmlRenderer();
        assertThat(renderer.renderAsXml(new ElementRepresentation("Abc"))).isEqualTo("<Abc />");
        assertThat(renderer.renderAsXml(new ElementRepresentation("Element"))).isEqualTo("<Element />");
    }
}

File: core/src/test/java/be/geoffrey/xmlweave/core/usecase/XmlRenderingTest.kt
package be.geoffrey.xmlweave.core.usecase

import be.geoffrey.xmlweave.core.usecase.xmlrendering.XmlRenderer
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class XmlRenderingTest {

    @Test
    fun simpleElementShouldBeRendered() {
        val renderer = XmlRenderer()
        assertThat(renderer.renderAsXml(Element("Abc"))).isEqualTo("<Abc />")
        assertThat(renderer.renderAsXml(Element("Element"))).isEqualTo("<Element />")
    }

    @Test
    fun simpleElementWithChildrenShouldBeRendered() {
        val renderer = XmlRenderer()
        val testElement = Element("Hello",
                children = listOf(
                        Element("World"),
                        Element("Wooooo")))

        assertThat(renderer.renderAsXml(testElement)).isEqualTo(trimmed(
                """
                <Hello>
                    <World />
                    <Wooooo />
                </Hello>
                """))
    }

    private fun trimmed(input: String): String {
        return input.trimIndent()
    }

}
